alphastats/dataset/dataset.py

```diff
@@ -199,6 +219,11 @@ def preprocess(
                 **kwargs,
             )
         )
+        (
```

Comment:
There will never be more keys than after the first execution during init. Both keys and values can change when mat changes. The idea of recalculating it is to avoid hickups when trying to look up a feature that has been removed during preprocessing.